features/sync/current_branch/feature_branch/merge_sync_strategy/no_push/enabled.feature

```diff
@@ -0,0 +1,45 @@
+Feature: no-push setting set to "true"
+
+  Background:
+    Given the current branch is a feature branch "feature"
+    And the commits
+      | BRANCH  | LOCATION | MESSAGE               |
+      | main    | local    | local main commit     |
+      |         | origin   | origin main commit    |
+      | feature | local    | local feature commit  |
+      |         | origin   | origin feature commit |
+    And Git Town setting "no-push" is "true"
+
+  Scenario: result
+    When I run "git-town sync"
+    Then it runs the commands
+      | BRANCH  | COMMAND                                 |
+      | feature | git fetch --prune --tags                |
+      |         | git checkout main                       |
+      | main    | git rebase origin/main                  |
+      |         | git push                                |
+      |         | git checkout feature                    |
+      | feature | git merge --no-edit --ff origin/feature |
+      |         | git merge --no-edit --ff main           |
+    And all branches are now synchronized
+    And feature branch is not pushed
```

Comment:
The fact that the feature branch is not pushed will be apparent in the list of commits in the table on the line right above: the tracking branch will not contain the local commits.

```suggestion
```